app/hooks/map/index.ts

```diff
@@ -552,7 +552,7 @@ export function useLegend({
     if (type === 'analysis' && subtype === 'analysis-adjust-planning-units') return ['wdpa-percentage', 'lock-in', 'lock-out', 'pugrid'];
     if (type === 'analysis') return ['wdpa-percentage', 'features', 'pugrid'];
     if (type === 'solutions') return ['frequency', 'solution', 'pugrid'];
-    if (type === 'published') return ['wdpa-percentage', 'frequency', 'features', 'pugrid'];
+    if (type === 'published') return ['frequency', 'features', 'pugrid'];
```

Comment:
I have a add a new type 'published' independent of the tabs to be able to customize what we see on the map.

Reply:
What we should do is the same we did with the layers. This "useMemo" to get the layers should come from outside, from where you are using the useLegend instead of from inside. That way we don't care if you are in the detail, the published page, or wherever you want. 

In other words, instead of passing type & subtype, you should pass directly an attribute called `layers` calculated where you use the `useLegend`